src/wallet/spend.cpp

```diff
@@ -1337,7 +1337,7 @@ util::Result<CreatedTransactionResult> CreateTransaction(
 
     auto res = CreateTransactionInternal(wallet, vecSend, change_pos, coin_control, sign);
     TRACE4(coin_selection, normal_create_tx_internal, wallet.GetName().c_str(), bool(res),
-           res ? res->fee : 0, res && res->change_pos.has_value() ? *res->change_pos : 0);
+           res ? res->fee : 0, res && res->change_pos.has_value() ? *res->change_pos : -1);
```

Comment:
```
(mocktwallet/spend.cpp:1339:5: runtime error: implicit conversion from type 'int' of value -1 (32-bit, signed) to type 'unsigned int' changed the value to 4294967295 (32-bit, unsigned)
    #0 0x5589c92846e6 in wallet::CreateTransaction(wallet::CWallet&, std::vector<wallet::CRecipient, std::allocator<wallet::CRecipient>> const&, std::optional<unsigned int>, wallet::CCoinControl const&, bool) src/wallet/spend.cpp:1339:5
    #1 0x5589c8d8617d in wallet::spend_tests::wallet_duplicated_preset_inputs_test::test_method() src/wallet/test/spend_tests.cpp:100:53
   ...

SUMMARY: UndefinedBehaviorSanitizer: implicit-integer-sign-change wallet/spend.cpp:1339:5 in 
```

Reply:
I think wrapping the trace parameter in `static_cast<int32_t>()` is the correct way to resolve this because `change_pos` is otherwise a `std::optional<unsigned int>`.  In tracing.md `change_pos` is defined as int32 so it should always be returned as a signed 32-bit int.

Reply:
.. or maybe `static_cast` is overkill, a cast with `int32_t()` seems to be what's recommended in developer-nodes.md .

Reply:
I had my casting on the wrong value - we need to cast the unsigned optional value. Fixed in [e701982](https://github.com/bitcoin/bitcoin/pull/29272/commits/e701982c741edd2182690ddf4628e4c079a5185e).